tests/k8s/k8s_test_utilities.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import time
+
+logger = logging.getLogger(__name__)
+
+WAIT_FOR_SYNC = 60 # time unit seconds
+
+
+def join_master(duthost, master_vip):
+    """
+    Joins DUT to Kubernetes master
+
+    Args:
+        duthost: DUT host object
+        master_vip: VIP of high availability Kubernetes master
+
+    If join fails, test will fail at the assertion to check_connected
+    """
+    logger.info("Joining DUT to Kubernetes master")
+    duthost.shell('sudo config kube server disable on')
+    duthost.shell('sudo config kube server ip {}'.format(master_vip))
+    duthost.shell('sudo config kube server disable off')
+    time.sleep(WAIT_FOR_SYNC)
+    assert check_connected(duthost)
+    
+
+def make_vip_unreachable(duthost, master_vip):
+    """
+    Makes Kubernetes master VIP unreachable from SONiC DUT by configuring iptables rules. Cleans preexisting iptables rules for VIP. 
+
+    Args:
+        duthost: DUT host object
+        master_vip: VIP of high availability Kubernetes master
+    """
+    logger.info("Making Kubernetes master VIP unreachable from DUT")
+    clean_vip_iptables_rules(duthost, master_vip)
+    duthost.shell('sudo iptables -A INPUT -s {} -j DROP'.format(master_vip))
+    duthost.shell('sudo iptables -A OUTPUT -d {} -j DROP'.format(master_vip))
+
+
+def make_vip_reachable(duthost, master_vip):
+    """
+    Makes Kubernetes master VIP reachable from SONiC DUT by removing any iptables rules associated with the VIP. 
+
+    Args:
+        duthost: DUT host object
+        master_vip: VIP of high availability Kubernetes master
+    """
+    logger.info("Making Kubernetes master VIP reachable from DUT")
+    clean_vip_iptables_rules(duthost, master_vip)
+
+
+def clean_vip_iptables_rules(duthost, master_vip):
+    """
+    Removes all iptables rules associated with the VIP.
+
+    Args:
+        duthost: DUT host object
+        master_vip: VIP of high availability Kubernetes master
+    """
+    iptables_rules = duthost.shell('sudo iptables -S | grep {} || true'.format(master_vip))["stdout"].split("\n")
+    logger.info('iptables rules: {}'.format(iptables_rules))
+    for line in iptables_rules:
+        if line: 
+            duthost.shell('sudo iptables -D {}'.format(line[2:]))
+
+
+def shutdown_all_api_server(k8shosts):
+    """
+    Shuts down the API server on each backend master server.
+
+    Args:
+        k8shosts:  Shortcut fixture for getting Kubernetes hosts
+    """
+    for i in range(1, len(k8shosts)):
+        k8shost = k8shosts['m{}'.format(i)]['host']
+        logger.info("Shutting down API Server on master node m{}".format(i))
+        k8shost.shutdown_api_server()
```

Comment:
If this operation need to be performed on all k8s master nodes, maybe it is better to create a new class like `K8sCluster` to represent the k8s masters cluster. The class should have attributes like self.k8smasters which holds all the k8s master instances. Then we can encapsulate such operations as method of K8sCluster. Eventually in test scripts, we can just call `k8scluster.shutdown_api_server()` or `k8scluster.start_api_server()`.

Reply:
That's a good point. I removed this function altogether because you helped me realize this test case should refrain from shutting down the API server on the master, in case other testbeds are using the master at the same time. I will keep the idea of a `K8sCluster` class in mind for the future.

Reply:
I ended up adding the class `K8sMasterCluster` to retrieve the VIP without indexing via `[ha]`. I also wrote the function to `shudown_all_api_server()` but it is not used for this test case